traffic_ops/traffic_ops_golang/user/user.go

```diff
@@ -508,3 +519,426 @@ func (user *TOUser) InsertQuery() string {
+	u.last_updated,`
+
+const readQuery = readBaseQuery + `
+	r.name as role
+FROM tm_user u
+LEFT JOIN tenant t ON u.tenant_id = t.id
+LEFT JOIN role r ON u.role = r.id
+`
+
+const legacyReadQuery = readBaseQuery + `
+	r.name AS rolename,
+	u.role
+FROM tm_user u
+LEFT JOIN tenant t ON u.tenant_id = t.id
+LEFT JOIN role r ON u.role = r.id
+`
+
+// this is necessary because tc.User doesn't read its RoleName field in sql
+// driver scans.
+type userGet struct {
+	RoleName *string `json:"rolename" db:"rolename"`
+	tc.User
+}
+
+func read(rows *sqlx.Rows) ([]tc.UserV50, error) {
+	if rows == nil {
+		return nil, errors.New("cannot read from nil rows")
+	}
+
+	users := []tc.UserV50{}
+	for rows.Next() {
+		var user tc.UserV50
+		if err := rows.StructScan(&user); err != nil {
+			return nil, fmt.Errorf("scanning UserV5 row: %w", err)
+		}
+		users = append(users, user)
+	}
+
+	return users, nil
+}
+
+func getMaxLastUpdated(where string, queryValues map[string]interface{}, tx *sqlx.Tx) (time.Time, error) {
+	query := selectMaxLastUpdatedQuery(where)
+	var t time.Time
+	rows, err := tx.NamedQuery(query, queryValues)
+	if err != nil {
+		return t, fmt.Errorf("query for max user last updated time: %w", err)
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		if err = rows.Scan(&t); err != nil {
+			return t, fmt.Errorf("scanning user max last updated time: %w", err)
+		}
+	}
+	return t, nil
+}
+
+// Get is the handler for GET requests made to /users.
+func Get(w http.ResponseWriter, r *http.Request) {
+	inf, userErr, sysErr, errCode := api.NewInfo(r, nil, nil)
+	tx := inf.Tx.Tx
+	if userErr != nil || sysErr != nil {
+		api.HandleErr(w, r, tx, errCode, userErr, sysErr)
+		return
+	}
+	defer inf.Close()
+
+	api.DefaultSort(inf, "username")
+	params := map[string]dbhelpers.WhereColumnInfo{
+		"id":       {Column: "u.id", Checker: api.IsInt},
+		"role":     {Column: "r.name"},
+		"tenant":   {Column: "t.name"},
+		"username": {Column: "u.username"},
+	}
+	params["company"] = dbhelpers.WhereColumnInfo{Column: "u.company"}
+	params["email"] = dbhelpers.WhereColumnInfo{Column: "u.email"}
+	params["fullName"] = dbhelpers.WhereColumnInfo{Column: "u.full_name"}
+	params["newUser"] = dbhelpers.WhereColumnInfo{Column: "u.new_user"}
+	params["city"] = dbhelpers.WhereColumnInfo{Column: "u.city"}
+	params["stateOrProvince"] = dbhelpers.WhereColumnInfo{Column: "u.state_or_province"}
+	params["country"] = dbhelpers.WhereColumnInfo{Column: "u.country"}
+	params["postalCode"] = dbhelpers.WhereColumnInfo{Column: "u.postal_code"}
+
+	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(inf.Params, params)
+	if len(errs) != 0 {
+		api.HandleErr(w, r, tx, http.StatusBadRequest, util.JoinErrs(errs), nil)
+		return
+	}
+
+	tenantIDs, err := tenant.GetUserTenantIDListTx(inf.Tx.Tx, inf.User.TenantID)
+	if err != nil {
+		api.HandleErr(w, r, tx, http.StatusInternalServerError, nil, fmt.Errorf("getting tenant list for user: %v\n", err))
```

Comment:
Errors should not contain newlines, also formatting for errors should use `%w` instead of `%v` so that it wraps